### ISSUE TITLE
PagedResourceCollection page() or size() functions don't keep the sort parameter present in the selfLink.href

### ISSUE THREAD

**Issue by grrenier** [lagoshny/ngx-hateoas-client#110]
Hello, 
In the `customPage()` function the previous sort parameter is being always deleted. Also if I change the page using `page()` the size should not be modified.

File: https://github.com/lagoshny/ngx-hateoas-client/tree/master/projects/ngx-hateoas-client/src/lib/model/resource/paged-resource-collection.ts

```
170 const requestUrl = new URL(this.selfLink.href);
171 requestUrl.searchParams.delete('page');
172 requestUrl.searchParams.delete('size');
173 requestUrl.searchParams.delete('sort');
```

**Comment by lagoshny** [comment]
Hi @grrenier ,

Thank you for the issue

I fixed this behavior in 3.6.1 version, please, check it.

**Comment by grrenier** [comment]
Hi @lagoshny ,
Nice fix!! it works as expected, thank you!
